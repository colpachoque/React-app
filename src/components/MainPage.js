import React from 'react';
import styles from './MainPage.css';
import MovieCard from './MovieCard';


class MainPage extends React.Component {
    render () {
        const { movies } = this.props;
        return (
            <div className={styles.container}>
                {movies.length 
                    ? movies.map((movie, index) => (
                        <MovieCard
                            key={index}
                            name={movie.name}
                            genre={movie.genre}
                            year={movie.year}
                        />
                    ))
                    : 'No moview found'
                }
            </div>
        );
    }
}

export default MainPage;