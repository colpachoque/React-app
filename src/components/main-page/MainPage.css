.container {
    display: flex;
    flex-wrap: wrap;
    width: 40%;
    margin: 0 auto;
}
