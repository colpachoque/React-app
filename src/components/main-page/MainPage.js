import React from 'react';
import styles from './MainPage.css';
import MovieCard from '../movie-card/MovieCard';

const movies = [
    {
        name: 'Kill Bill',
        genre: 'Action and adventure',
        year: 2004
    },
    {
        name: 'Kill Bill',
        genre: 'Action and adventure',
        year: 2004
    }, 
    {
        name: 'Kill Bill',
        genre: 'Action and adventure',
        year: 2004
    }, 
    {
        name: 'Kill Bill',
        genre: 'Action and adventure',
        year: 2004
    },
    {
        name: 'Kill Bill',
        genre: 'Action and adventure',
        year: 2004
    },
    {
        name: 'Kill Bill',
        genre: 'Action and adventure',
        year: 2004
    },
    {
        name: 'Kill Bill',
        genre: 'Action and adventure',
        year: 2004
    }
];

class MainPage extends React.Component {
    render () {
        return (
            <div className={styles.container}>
                {movies.length 
                    ? movies.map((movie, index) => (
                        <MovieCard
                            key={index}
                            name={movie.name}
                            genre={movie.genre}
                            year={movie.year}
                        />
                    ))
                    : 'No moview found'
                }
            </div>
        );
    }
}

export default MainPage;