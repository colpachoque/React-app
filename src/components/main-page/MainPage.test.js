import React from "react";
import MainPage from "./MainPage";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
describe("Main Page component", () => {
  let movie;

  beforeAll(() => {
    movie = {
      name: "The best movie ever",
      genre: "Drama",
      year: 1994
    };
  });

  it("should be render correctly", () => {
    const component = shallow(
      <MainPage name={movie.name} genre={movie.genre} year={movie.year} />
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
