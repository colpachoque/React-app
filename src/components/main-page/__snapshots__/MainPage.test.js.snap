// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page component should be render correctly 1`] = `
<div
  className="container"
/>
`;
